### ISSUE TITLE
weight decay = 0 for BN parameters

### ISSUE THREAD

**Issue by wesleywesleyteng** [JiahuiYu/slimmable_networks#15]
Hi, Jiahui,

I notice that in function "get_optimizer()" of train.py, weight_decay are set as 0 for BN parameter(fc bias as well!?), which is not like what we commonly do. As far as I know, we usually apply a equal weight_decay for all model parameters (correct me if i'm wrong). I am curious about what is the purpose of doing this? Many thanks!
![Selection_497](https://user-images.githubusercontent.com/50227470/57121972-63a46a00-6dad-11e9-9e23-b2c7e0ddf562.png)

**Comment by VectorYoung** [comment]
@wesleywesleyteng 
Hi, you can refer to the [MobileNet](https://arxiv.org/pdf/1704.04861.pdf) paper. That's a common setting for low complexity networks.

**Comment by JiahuiYu** [comment]
@VectorYoung Thanks for your help to @wesleywesleyteng .

**Comment by wesleywesleyteng** [comment]
@VectorYoung,
Hi, I agree with you  there should be no weight_decay on depthwise filters.
But what I mean here is those weight_decay values are also set as 0 on batchNorm parameters.
Here is a simple print log when I run model 's_mobilenet_v1'
![image](https://user-images.githubusercontent.com/50227470/57211186-f5acac80-7011-11e9-854c-1a4eb609486e.png)
Thank you!

**Comment by VectorYoung** [comment]
@wesleywesleyteng 
Hi, I see your point. I agree with you that there should be weight decay for USBatchNorm's weight. I am not sure why it is set to 0. @JiahuiYu

**Comment by JiahuiYu** [comment]
@VectorYoung @wesleywesleyteng Sorry for the late reply.

Setting weight decay of Batch Normalization to zero is found **very important** for (slimmable) object detection experiments. No weight decaying on BN is commonly used in detectron, mmdetection and other frameworks (I remember it is also used in paper Group Normalization). For image classification tasks, I have not compared the performance. Since the central idea of our work is modifying/adapting BN, enabling weight decaying on BN sounds very weird to me.

If any of you have related experiments (weight decay vs. no weight decay), it would be great if you can report here. Thanks!
